### ISSUE TITLE
error on compile :assets_audio_player:compileDebugKotlin'.

### ISSUE THREAD

**Issue by robertoltrocha** [florent37/Flutter-AssetsAudioPlayer#399]
Hi,
I can not compile my project because the issue below, someone could help me please?
```
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\assets_audio_player-2.0.13+1\android\src\main\kotlin\com\github\florent37\assets_audio_player\Player.kt: (542, 5): 'handleMessage' overrides nothing
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\assets_audio_player-2.0.13+1\android\src\main\kotlin\com\github\florent37\assets_audio_player\Player.kt: (543, 29): Type mismatch: inferred type is Message? but Message was expected
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\assets_audio_player-2.0.13+1\android\src\main\kotlin\com\github\florent37\assets_audio_player\notification\NotificationActionReceiver.kt: (12, 85): Type mismatch: inferred type is String? but String was expected
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\hosted\pub.dartlang.org\assets_audio_player-2.0.13+1\android\src\main\kotlin\com\github\florent37\assets_audio_player\notification\NotificationActionReceiver.kt: (30, 39): Type mismatch: inferred type is Intent? but Intent was expected

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':assets_audio_player:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

BUILD FAILED in 26s
Exception: Gradle task assembleDebug failed with exit code 1

`assets_audio_player: ^2.0.13+1`

```
Flutter is already up to date on channel stable
Flutter 1.22.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 8874f21e79 (13 days ago) • 2020-10-29 14:14:35 -0700
Engine • revision a1440ca392
Tools • Dart 2.10.3
```

**Comment by imSaharukh** [comment]
facing the same problem

**Comment by alberduris** [comment]
Same issue here with Flutter 1.22.4.

This is the output for `flutter run --debug`
```
Launching lib/main.dart on Android SDK built for x86 in debug mode...
e: /home/user/development/flutter/.pub-cache/hosted/pub.dartlang.org/assets_audio_player-2.0.13+1/android/src/main/kotlin/com/github/florent37/assets_audio_player/notification/NotificationActionReceiver.kt: (30, 39): Type mismatch: inferred type is Intent? but Intent was expected
                                                                        
FAILURE: Build failed with an exception.                                
                                                                        
* What went wrong:                                                      
Execution failed for task ':assets_audio_player:compileDebugKotlin'.    
> Compilation error. See log for more details                           
                                                                        
* Try:                                                                  
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
                                                                        
* Get more help at https://help.gradle.org                              
                                                                        
BUILD FAILED in 15s                                                     
Running Gradle task 'assembleDebug'...                                  
Running Gradle task 'assembleDebug'... Done                        16.2s
Exception: Gradle task assembleDebug failed with exit code 1
```
This is the output for `flutter doctor -v`

```
[✓] Flutter (Channel stable, 1.22.4, on Linux, locale en_GB.UTF-8)
    • Flutter version 1.22.4 at /home/ablanco061/development/flutter
    • Framework revision 1aafb3a8b9 (9 days ago), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /home/ablanco061/Android/Sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /usr/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.9.1+1-Ubuntu-0ubuntu1.18.04)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/linux#android-setup for more details.

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[✓] Connected device (1 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86 • Android 10 (API 29) (emulator)

! Doctor found issues in 2 categories.
```

**Comment by kalismeras61** [comment]
Could you follow below steps ?

flutter clean
> rm -rf pubspec.lock .packages .flutter-plugins
> flutter pub cache repair
> flutter packages get

**Comment by alberduris** [comment]
<details>
<summary>flutter clean:</summary>

```
Deleting build...                                                  191ms
Deleting .dart_tool...                                               1ms
Deleting Generated.xcconfig...                                       1ms
Deleting flutter_export_environment.sh...                            0ms
Deleting .flutter-plugins-dependencies...                            1ms
Deleting .flutter-plugins...                                         0ms
```

</details>

<details>
<summary>flutter pub cache repair:</summary>

```
Reinstalled 399 packages.
Reactivating devtools 0.9.3+4...
Precompiling executables...
Precompiled devtools:devtools.
Installed executable devtools.
Reactivated 1 package.
```

</details>

Exactly the same issue.

It is also interesting to note that the import line is greyed out, indicating that I could run the app without expliciting the import. 

`import 'package:assets_audio_player/assets_audio_player.dart';`

So this line is not throwing an error even if I remove the import line

` final assetsAudioPlayer = AssetsAudioPlayer();`

**Comment by nagarwa2** [comment]
Facing same issue. When this will be resolved ?

**Comment by yelkamel** [comment]
@nagarwa2 you have to change your gradle version to this:
from 
`classpath 'com.android.tools.build:gradle:4.1.1'`
to
` classpath 'com.android.tools.build:gradle:3.5.3'`

Merci @florent37 pour ce package, quand t'as le temps essaye de voir le soucis avec la monter de version . ;)

**Comment by alberduris** [comment]
That's not a solution. I cannot do that as I rely on 4.1.1 version of gradle for other parts of the application.

**Comment by harshmittal2810** [comment]
I am still facing the same problem, @florent37 Can you please update the library. It's very crucial, 
Thanks for your help in advance.

**Comment by harshmittal2810** [comment]
> @nagarwa2 you have to change your gradle version to this:
> from
> `classpath 'com.android.tools.build:gradle:4.1.1'`
> to
> ` classpath 'com.android.tools.build:gradle:3.5.3'`
> 
> Merci @florent37 pour ce package, quand t'as le temps essaye de voir le soucis avec la monter de version. On ne peux plus faire d'application Android en App Bundle sans cette mise à jours, on est obligé de faire un apk ;)

This worked but as we need latest version its not recommended.

**Comment by leidig54** [comment]
I downgraded to 2.0.10 to solve this.

**Comment by jjsebastianfuertes** [comment]
any solution so far implementing classpath 'com.android.tools.build:gradle:4.1.1'?

**Comment by EgHubs** [comment]
any news?

**Comment by UsamaSarwar** [comment]
**[Solved]** ✨
Goto file `android/build.gradle`
```gradle
// classpath 'com.android.tools.build:gradle:4.1.0'
classpath 'com.android.tools.build:gradle:3.5.3'
```
Goto file `pubspec.yaml`
```yaml
dependencies:
  assets_audio_player: ^2.0.10
```
_This will help you in building the app release._
_This might save your nights_ ♥

**Comment by EgHubs** [comment]
Thanks for your reply, but I already solved that by that solution as someone already said it before, I mean a solution with the same 
`classpath 'com.android.tools.build:gradle:4.1.1'`

**Comment by DomingoMG** [comment]
I have the same problem.

You should update the repository, this way we are limited.

`classpath 'com.android.tools.build:gradle:4.1.1'`
NOT 
`classpath 'com.android.tools.build:gradle:3.5.3'`

**Comment by bibali1980** [comment]
1- download the package from here : https://github.com/florent37/Flutter-AssetsAudioPlayer
2- extract it in the same folder of your app and rename the extracted folder to : assets_audio_player
3- in this file : assets_audio_player\pubspec.yaml  
    *comment: #assets_audio_player_web: ^2.0.13+2
    *comment:       #web:
                                #default_package: assets_audio_player_web
4- in this file: assets_audio_player\android\src\main\kotlin\com\github\florent37\assets_audio_player\notification\NotificationActionReceiver.kt
   *just add !! at the end of this line: var intent : Intent = context.packageManager.getLaunchIntentForPackage(context.packageName)
    *it well be : var intent : Intent = context.packageManager.getLaunchIntentForPackage(context.packageName)!!
5- in your app pubspec.yaml add under dependencies:
  #assets_audio_player: ^2.0.13+1
  assets_audio_player:
    path: ../assets_audio_player/

**Comment by ianldgs** [comment]
> 1- download the package from here : https://github.com/florent37/Flutter-AssetsAudioPlayer
> 2- extract it in the same folder of your app and rename the extracted folder to : assets_audio_player
> 3- in this file : assets_audio_player\pubspec.yaml
> *comment: #assets_audio_player_web: ^2.0.13+2
> *comment: #web:
> #default_package: assets_audio_player_web
> 4- in this file: assets_audio_player\android\src\main\kotlin\com\github\florent37\assets_audio_player\notification\NotificationActionReceiver.kt
> *just add !! at the end of this line: var intent : Intent = context.packageManager.getLaunchIntentForPackage(context.packageName)
> *it well be : var intent : Intent = context.packageManager.getLaunchIntentForPackage(context.packageName)!!
> 5- in your app pubspec.yaml add under dependencies:
> #assets_audio_player: ^2.0.13+1
> assets_audio_player:
> path: ../assets_audio_player/

Why not make a PR on this repository with this change you're describing?

**Comment by robertoltrocha** [comment]
Hi,
Please,
any official solution? new version?
For now, I had to remove the plugin due to the priority need for the update to classpath 'com.android.tools.build:gradle:4.1.1'
Thank you

**Comment by kalismeras61** [comment]
Please latest version with commit id. 

```
assets_audio_player:
git:
url: https://github.com/florent37/Flutter-AssetsAudioPlayer.git
ref: b49a4360f6a7f182d4a6e23ebcdbf43ad570913b

```

**Comment by kalismeras61** [comment]
I am closing for now. Please let me know, if it is not working.

**Comment by robertoltrocha** [comment]
Hi,
I installed the plugin again and I still have the same problem.

```
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-cba4cb34e9263f96dcaa71118b39590630f077a4\android\src\main\kotlin\com\github\florent37\assets_audio_player\Player.kt: (542, 5): 'handleMessage' overrides nothing
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-cba4cb34e9263f96dcaa71118b39590630f077a4\android\src\main\kotlin\com\github\florent37\assets_audio_player\Player.kt: (543, 29): Type mismatch: inferred type is Message? but Message was expected
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-cba4cb34e9263f96dcaa71118b39590630f077a4\android\src\main\kotlin\com\github\florent37\assets_audio_player\notification\CustomMediaButtonReceiver.kt: (15, 35): Type mismatch: inferred type is String? but String was expected
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-cba4cb34e9263f96dcaa71118b39590630f077a4\android\src\main\kotlin\com\github\florent37\assets_audio_player\notification\NotificationActionReceiver.kt: (12, 85): Type mismatch: inferred type is String? but String was expected

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':assets_audio_player:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 6s
Exception: Gradle task assembleDebug failed with exit code 1
```

**Comment by kalismeras61** [comment]
Delete the pubspec.lock file then run the command flutter pub get again.

**Comment by robertoltrocha** [comment]
Hi @kalismeras61, 
Unsuccessfully.
I am using:     **classpath 'com.android.tools.build:gradle:4.1.1'**
Steps:
- Delete pubspec.lock
- flutter clean
- invalidate and restart
- flutter pub get
- run

```
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-cba4cb34e9263f96dcaa71118b39590630f077a4\android\src\main\kotlin\com\github\florent37\assets_audio_player\Player.kt: (542, 5): 'handleMessage' overrides nothing
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-cba4cb34e9263f96dcaa71118b39590630f077a4\android\src\main\kotlin\com\github\florent37\assets_audio_player\Player.kt: (543, 29): Type mismatch: inferred type is Message? but Message was expected
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-cba4cb34e9263f96dcaa71118b39590630f077a4\android\src\main\kotlin\com\github\florent37\assets_audio_player\notification\CustomMediaButtonReceiver.kt: (15, 35): Type mismatch: inferred type is String? but String was expected
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-cba4cb34e9263f96dcaa71118b39590630f077a4\android\src\main\kotlin\com\github\florent37\assets_audio_player\notification\NotificationActionReceiver.kt: (12, 85): Type mismatch: inferred type is String? but String was expected

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':assets_audio_player:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 13s
Exception: Gradle task assembleDebug failed with exit code 1
```

**Comment by kalismeras61** [comment]
There is a problem with your pub cache . Can you remove all assets player in pub cache folder ? and then run flutter pub get

**Comment by robertoltrocha** [comment]
Hi @kalismeras61, 

same issue yet.

I removed everything in my folder cache  and subfolders with relationship assests_audio. 

> C:\Users\rocha\AppData\Roaming\Pub\Cache\

and after that:
Steps:
-Delete pubspec.lock
-flutter clean
-invalidate and restart
-flutter pub get
-run

```
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-cba4cb34e9263f96dcaa71118b39590630f077a4\android\src\main\kotlin\com\github\florent37\assets_audio_player\Player.kt: (542, 5): 'handleMessage' overrides nothing
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-cba4cb34e9263f96dcaa71118b39590630f077a4\android\src\main\kotlin\com\github\florent37\assets_audio_player\Player.kt: (543, 29): Type mismatch: inferred type is Message? but Message was expected
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-cba4cb34e9263f96dcaa71118b39590630f077a4\android\src\main\kotlin\com\github\florent37\assets_audio_player\notification\CustomMediaButtonReceiver.kt: (15, 35): Type mismatch: inferred type is String? but String was expected
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-cba4cb34e9263f96dcaa71118b39590630f077a4\android\src\main\kotlin\com\github\florent37\assets_audio_player\notification\NotificationActionReceiver.kt: (12, 85): Type mismatch: inferred type is String? but String was expected

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':assets_audio_player:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 12s
Exception: Gradle task assembleDebug failed with exit code 1

```

**Comment by kalismeras61** [comment]
Can you show how are you import to package in your pubspec.yaml file ?

**Comment by robertoltrocha** [comment]
Sure.
There is my all pubspec.yaml

```
environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  flutter_localizations:
    sdk: flutter
  url_launcher: 5.7.10
  cached_network_image: 2.5.0
  http: 0.12.2
  awesome_dialog: 1.3.1
  #flutter_masked_text: 0.8.0
  card_settings: ^1.16.2
  flutter_launcher_icons: 0.8.1
  flutter_svg: 0.19.1
  #soundpool: 1.1.3
  progress_indicators: 0.1.5 #texto pulando
  auto_size_text: 2.1.0
  shared_preferences: 0.5.12+4
  expandable: 4.1.4
  #assets_audio_player: 2.0.13+1
  assets_audio_player:
    git:
      url: https://github.com/florent37/Flutter-AssetsAudioPlayer.git
      ref: master

  grouped_list: 3.7.1
  flutter_slidable: 0.5.7
  flutter_staggered_animations: 0.1.2
  flutter_barcode_scanner: 1.0.2
  permission_handler: 5.0.1+1
  video_player: 1.0.1
  convex_bottom_bar: 2.6.0
  mask_text_input_formatter: 1.2.1
  intl: 0.16.1
  scoped_model: 1.1.0
  email_validator: 1.0.6
  flushbar: 1.10.4
  flutter_phoenix: 0.1.0
  animated_splash: 1.0.0
  photo_view: 0.10.3
  flutter_spinkit: 4.1.2+1
  flutter_typeahead: 2.0.0
  flutter_login: 1.0.14
  json_serializable: 3.5.1
  argon_buttons_flutter: 1.0.6
  #pull_to_refresh: 1.6.3
  rxdart: 0.25.0
  flappy_search_bar: 1.7.2
  package_info: 0.4.3+4
  image_picker: 0.6.7+22
  #excel: 1.1.5
  percent_indicator: 2.1.9
  badges: 1.1.6
  camerawesome: 0.2.1+2
  #dynamic_theme: 1.0.1
  flutter_cached_pdfview: ^0.3.5
  syncfusion_flutter_charts: 18.4.35
  #syncfusion_flutter_gauges: 18.4.34
  syncfusion_flutter_calendar: 18.4.39
  syncfusion_localizations: 18.4.39
  syncfusion_flutter_sliders: 18.4.39-beta
  syncfusion_flutter_core: 18.4.39
  syncfusion_flutter_datepicker: ^18.4.39-beta
  #getwidget: 1.2.4
  font_awesome_flutter: 8.11.0
  #animated_text_kit: 2.5.4
  #printing: 3.7.1
  #modal_bottom_sheet: ^1.0.0+1
  #share: ^0.6.5+4
  #animations 1.1.2
  #sqflite 1.3.2+1;
  sliding_up_panel: 1.0.2
  #timeline_tile
  #curved_navigation_bar 0.3.4
  #flutter_screenutil 3.2.0
  flutter_neumorphic: 3.0.3
  google_fonts: ^1.1.2
  #carousel_slider:  3.0.0
  dropdown_search: ^0.4.8
  random_color: ^1.0.5
  #map_launcher: ^1.1.1
  #geolocator: ^6.1.13
  #google_maps_flutter: ^1.0.6
  #geocoding: ^1.0.5
  #geocoder: ^0.2.1
  workmanager: ^0.2.3
  jiffy: ^3.0.1
  wakelock: ^0.2.1+1
  open_file: ^3.0.3
  flutter_cache_manager: ^2.1.1
  device_info: ^1.0.0
  in_app_update: ^1.1.15
  #sqflite: ^1.3.0
  #archive: ^2.0.13
  #pin_code_text_field 1.7.1
  #local_auth: ^0.6.3+4


dev_dependencies:

  icapps_license: 0.0.6
  flutter_test:
    sdk: flutter
  build_runner: 1.10.11
  flutter_driver:
    sdk: flutter
  #pedantic: 1.8.0


flutter_icons:
  android: "launcher_icon"
  ios: true
  image_path: "assets/images/logo_3.jpg"

flutter:
  uses-material-design: true
  fonts:
    - family: Samsung
      fonts:
        - asset: fonts/SamsungSans-Bold.ttf
          weight: 700

  assets:
    - assets/images/
    - assets/sounds/
    - assets/images/leds/
  #   - images/a_dot_ham.jpeg
```

**Comment by kalismeras61** [comment]
I did some changes in another branch. Can you test with this. Please do flutter clean remove pubspec.lock file and import package run flutter pub get. 
```
 assets_audio_player:
    git:
      url: https://github.com/florent37/Flutter-AssetsAudioPlayer.git
      ref: fix-building
```

**Comment by robertoltrocha** [comment]
Hi @kalismeras61,

same issue yet.

I removed **again** everything in my folder cache and subfolders with relationship assests_audio.

C:\Users\rocha\AppData\Roaming\Pub\Cache\

and after that:
Steps:
-Delete pubspec.lock
-flutter clean
-invalidate and restart
-flutter pub get
-run

```
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-a8e5083f10b2aa8482f586d00af1349ca19c183a\android\src\main\kotlin\com\github\florent37\assets_audio_player\Player.kt: (542, 5): 'handleMessage' overrides nothing
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-a8e5083f10b2aa8482f586d00af1349ca19c183a\android\src\main\kotlin\com\github\florent37\assets_audio_player\Player.kt: (543, 29): Type mismatch: inferred type is Message? but Message was expected
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-a8e5083f10b2aa8482f586d00af1349ca19c183a\android\src\main\kotlin\com\github\florent37\assets_audio_player\notification\CustomMediaButtonReceiver.kt: (15, 35): Type mismatch: inferred type is String? but String was expected
e: C:\Users\rocha\AppData\Roaming\Pub\Cache\git\Flutter-AssetsAudioPlayer-a8e5083f10b2aa8482f586d00af1349ca19c183a\android\src\main\kotlin\com\github\florent37\assets_audio_player\notification\NotificationActionReceiver.kt: (12, 85): Type mismatch: inferred type is String? but String was expected

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':assets_audio_player:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 26s
Exception: Gradle task assembleDebug failed with exit code 1
```

**Comment by kalismeras61** [comment]
What is your minSdkVersion ?

**Comment by robertoltrocha** [comment]
Hi, @kalismeras61 ,
I was using:
- minSdkVersion 22
- targetSdkVersion 30
- compileSdkVersion 30

**good news: I changed evertthing to 29 and work well with debug mode.**
I will need to do more tests.
 - publish
 - do a test with mobiles 
 - I am using S-10+ with android 10
 - to do a test setting with:
-- minSdkVersion 22
-- targetSdkVersion 29
-- compileSdkVersion 29

**Comment by kalismeras61** [comment]
Cool . I  am glad to hear that. Please do test and let me know.

**Comment by robertoltrocha** [comment]
Hi, @kalismeras61,

I published it with version 22 and 29 and it passed, but I will only be able to do tests on the physical mobile tomorrow.
I returned the configuration to minSDK22 and targetSDK30 and again I had the same error. It seems that the problem is when we use the sdk30.

**Comment by offline-first** [comment]
Same issue with targetSdkVersion 30

**Comment by robertoltrocha** [comment]
Hi @offline-first,
I'm temporarily using targetSDK 29 and not 30. With that, everything worked.

**Comment by abbasDD** [comment]
also, work by changing         
classpath 'com.android.tools.build:gradle:4.1.2'
to
classpath 'com.android.tools.build:gradle:3.5.3'
but downgrade isn't the best solution, @florent37 kindly make it available for the latest versions also, thanks in advance

**Comment by dfields186** [comment]
I just followed all the steps robertoltrocha outlined and I'm still failing to compile. Is there any working solution to this issue?  If not, I'm going to just need to look for a different flutter package which plays mp3s.

Thanks!

**Comment by CheddersGroup** [comment]
I had this same problem.

what i did was to downgrade classpath from

classpath 'com.android.tools.build:gradle:4.1.1'
down to
classpath 'com.android.tools.build:gradle:3.5.3'

i my project build

**Comment by Amr1440** [comment]
Same issue and non of the above solutions worked

**Comment by Sherlock-White** [comment]
I tried to change gradle's version from 4.4.1 to 3.5.3, but it did not work.
And I also tried to change  assets_audio_player's version to what is mentioned above, but it still did not work.
I found my targetSdkVersion is bound to flutter.targetSdkVersion, so I tried to uninstall my flutter and downloaded **flutter_windows_2.10.3-stable**, **and it works!!** (although it still have some warnings about assets_audio_player, but now I can run the app on my pad for testing).
Hope this may work for you.
